app.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+import logging
+import os
+import typing
+
+import boto3
+from django.db import transaction  # type: ignore
+from openstates.cli.reports import generate_session_report
+
+
+logger = logging.getLogger("openstates")
+s3_client = boto3.client('s3')
+
+
+def process_upload_function(event, context):
+    """
+    Process a file upload.
+    """
+
+    # Get the uploaded file's information
+    bucket = event['Records'][0]['s3']['bucket']['name']  # Will be `my-bucket`
```

Comment:
Should we be checking more than the first value of `event["Records"]`? Is it possible that these sub-keys might not exist?

Reply:
This snippet is from the [AWS documentation ](https://docs.aws.amazon.com/lambda/latest/dg/with-s3-example.html) on how to retrieve the current metadata of the uploaded file, however, I can dig in on this to see if we have any reason to do otherwise